src/measure_theory/lp_space.lean

```diff
@@ -632,4 +640,168 @@ end
+
+lemma ae_measurable (f : Lp E p μ) : ae_measurable f μ := f.val.ae_measurable
+
+lemma mem_ℒp (f : Lp E p μ) : mem_ℒp f p μ := ⟨ae_measurable f, f.prop⟩
+
+/-- The norm in Lp space (which verifies the triangle inequality only for `1 ≤ p`) -/
+def Lp_norm (f : Lp E p μ) : ℝ := ennreal.to_real (snorm f p μ)
+
+instance : has_norm (Lp E p μ) := { norm := λ f, Lp_norm f }
+
+@[simp] lemma norm_eq_Lp_norm {f : Lp E p μ} : ∥f∥ = Lp_norm f := rfl
+
+lemma coe_fn_zero : ⇑(0 : Lp E p μ) =ᵐ[μ] 0 := ae_eq_fun.coe_fn_zero
+
+@[simp] lemma Lp_norm_zero : Lp_norm (0 : Lp E p μ) = 0 :=
+by simp [Lp_norm, snorm_congr_ae ae_eq_fun.coe_fn_zero, snorm_zero]
+
+lemma mem_Lp_const (α) [measurable_space α] (μ : measure α) (c : E) [finite_measure μ] :
+  @ae_eq_fun.const α _ _ μ _ c ∈ Lp E p μ :=
+(mem_ℒp_const c).snorm_mk_lt_top
+
+lemma Lp_norm_eq_zero_iff {f : Lp E p μ} (hp : 0 < p) : Lp_norm f = 0 ↔ f = 0 :=
+begin
+  refine ⟨λ hf, _, λ hf, by simp [hf]⟩,
+  rw [Lp_norm, ennreal.to_real_eq_zero_iff] at hf,
+  cases hf,
+  { rw snorm_eq_zero_iff (ae_measurable f) hp.ne.symm at hf,
+    exact subtype.eq (ae_eq_fun.ext (hf.trans ae_eq_fun.coe_fn_zero.symm)), },
+  { exact absurd hf (snorm_ne_top f), },
+end
+
+lemma coe_fn_neg {f : Lp E p μ} : ⇑(-f) =ᵐ[μ] -f := ae_eq_fun.coe_fn_neg _
+
+@[simp] lemma Lp_norm_neg {f : Lp E p μ} : Lp_norm (-f) = Lp_norm f :=
+by rw [Lp_norm, Lp_norm, snorm_congr_ae coe_fn_neg, snorm_neg]
+
+lemma coe_fn_add {f g : Lp E p μ} : ⇑(f + g) =ᵐ[μ] f + g := ae_eq_fun.coe_fn_add _ _
+
+lemma coe_fn_sub {f g : Lp E p μ} : ⇑(f - g) =ᵐ[μ] f - g := ae_eq_fun.coe_fn_sub _ _
+
+instance : has_dist (Lp E p μ) := { dist := λ f g, Lp_norm (f - g) }
+
+@[simp] lemma dist_eq_Lp_norm_sub {f g : Lp E p μ} : dist f g = Lp_norm (f - g) := rfl
+
+private lemma dist_triangle' [hp : fact (1 ≤ p)] (f g h : Lp E p μ) :
+  dist f h ≤ dist f g + dist g h :=
+begin
+  simp only [dist, Lp_norm],
+  rw ←ennreal.to_real_add (snorm_ne_top (f - g)) (snorm_ne_top (g - h)),
+  suffices h_snorm : snorm ⇑(f - h) p μ ≤ snorm ⇑(f - g) p μ + snorm ⇑(g - h) p μ,
+  { rwa ennreal.to_real_le_to_real (snorm_ne_top (f - h)),
+    exact ennreal.add_ne_top.mpr ⟨snorm_ne_top (f - g), snorm_ne_top (g - h)⟩, },
+  have h_add : (f - h) = f - g + (g - h), by abel,
+  rw [h_add, snorm_congr_ae coe_fn_add],
+  exact snorm_add_le (ae_measurable (f - g)) (ae_measurable (g - h)) hp,
+end
+
+instance [hp : fact (1 ≤ p)] : metric_space (Lp E p μ) :=
+{ dist_self := λ _, by simp,
+  dist_comm := λ _ _, by { simp only [dist], rw [←Lp_norm_neg, neg_sub] },
+  dist_triangle := dist_triangle',
+  eq_of_dist_eq_zero := λ _ _ h, by simpa [Lp_norm_eq_zero_iff (ennreal.zero_lt_one.trans_le hp),
+    sub_eq_zero] using h, }
+
+instance [fact (1 ≤ p)] : normed_group (Lp E p μ) := { dist_eq := λ _ _, by simp [dist, norm] }
+
+section normed_space
+
+variables {𝕜 : Type*} [normed_field 𝕜] [normed_space 𝕜 E]
+
+lemma mem_Lp_const_smul (c : 𝕜) (f : Lp E p μ) : c • f.val ∈ Lp E p μ :=
```

Comment:
You should avoid ` f.val`, and use the coercion instead, as this is the normal form we use.